sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -124,10 +126,75 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
       extraConf: util.Map[String, String]): Unit = {
     try {
       val conf = new RapidsConf(extraConf.asScala.toMap)
+      val cudfPropertiesFileName = "cudf-java-version-info.properties"
+      val pluginPropertiesFileName = "rapids4spark-version-info.properties"
       if (conf.shimsProviderOverride.isDefined) {
         ShimLoader.setSparkShimProviderClass(conf.shimsProviderOverride.get)
       }
 
+      // Compare if the cudf version mentioned in the classpath is equal to the version which plugin
+      // expects. If there is a version mismatch, throw error. This check can be disabled by
+      // setting this config spark.rapids.cudfVersionOverride=true
+      val props = new Properties
+      val cudfClassLoader = classOf[ai.rapids.cudf.ColumnVector].getClassLoader
+      val cudfProperties = cudfClassLoader.getResourceAsStream(cudfPropertiesFileName)
+      if (cudfProperties == null) {
+        if (!conf.cudfVersionOverride) {
+          throw new RuntimeException("Could not find properties file in the cudf jar." +
```

Comment:
This should use `cudfPropertiesFileName` in the error message so we know which file it cannot find.  It also should place this message in a local variable so it can be reused in both the exception path and the log path since it's the same message.

Also should not use "plugin" but "RAPIDS Accelerator" per @revans2 earlier comment.